streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -177,63 +177,96 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
         streamName = (streamName == null) ? this.name : streamName;
-        topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(null, streamName), keySerde, valSerde), this.name);
+        topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(null, mapper, streamName), keySerde, valSerde), this.name);
     }
 
-
     @Override
     public void writeAsText(String filePath) {
-        writeAsText(filePath, null, null, null);
+        writeAsText(filePath, null, null, null, null);
     }
 
     @Override
     public void writeAsText(String filePath, String streamName) {
-        writeAsText(filePath, streamName, null, null);
+        writeAsText(filePath, streamName, null, null, null);
     }
 
-
     @Override
     public void writeAsText(String filePath, Serde<K> keySerde, Serde<V> valSerde) {
-        writeAsText(filePath, null, keySerde, valSerde);
+        writeAsText(filePath, null, keySerde, valSerde, null);
     }
 
-    /**
-     * @throws TopologyBuilderException if file is not found
-     */
     @Override
     public void writeAsText(String filePath, String streamName, Serde<K> keySerde, Serde<V> valSerde) {
+        writeAsText(filePath, streamName, keySerde, valSerde, null);
+    }
+
+    @Override
+    public void writeAsText(final String filePath, final KeyValueMapper<? super K, ? super V, String> mapper) {
+        writeAsText(filePath, null, null, null, mapper);
+    }
+
+    @Override
+    public void writeAsText(final String filePath, final String streamName, final KeyValueMapper<? super K, ? super V, String> mapper) {
+        writeAsText(filePath, streamName, null, null, mapper);
+    }
+
+    @Override
+    public void writeAsText(final String filePath, final Serde<K> keySerde, final Serde<V> valSerde, final KeyValueMapper<? super K, ? super V, String> mapper) {
+        writeAsText(filePath, null, keySerde, valSerde, mapper);
+    }
+
+    @Override
+    public void writeAsText(final String filePath, String streamName, final Serde<K> keySerde, final Serde<V> valSerde, final KeyValueMapper<? super K, ? super V, String> mapper) {
         Objects.requireNonNull(filePath, "filePath can't be null");
         if (filePath.trim().isEmpty()) {
             throw new TopologyBuilderException("filePath can't be an empty string");
         }
         String name = topology.newName(PRINTING_NAME);
         streamName = (streamName == null) ? this.name : streamName;
         try {
-            PrintWriter printWriter = null;
-            printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
-            topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(printWriter, streamName), keySerde, valSerde), this.name);
+            PrintWriter printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
+            topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(printWriter, mapper, streamName), keySerde, valSerde), this.name);
```

Comment:
note that I just do it in local and not yet to PR, sorry for an ambiguous statement